packages/waku/src/lib/rsc/handler-dev.ts

```diff
@@ -36,27 +45,40 @@ export function createHandler<
     throw new Error('prehook is required if posthook is provided');
   }
   const configPromise = resolveConfig(options.config || {});
-
   const vitePromise = configPromise.then(async (config) => {
-    const viteServer = await viteCreateServer({
-      base: config.basePath,
-      optimizeDeps: {
-        include: ['react-server-dom-webpack/client'],
-        exclude: ['waku'],
+    const resolvedViteConfig = await resolveViteConfig({}, 'serve');
+    const mergedViteConfig = await mergeViteConfig(
+      {
+        ...resolvedViteConfig,
+
+        plugins: resolvedViteConfig.plugins.filter(
+          (plugin) => !plugin.name.startsWith('vite:'),
+        ),
       },
```

Comment:
What happens if we don't have this? Why do we need to remove `vite:` plugins?

Reply:
React refresh would fail